thread/thread_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package thread
+
+import (
+	"encoding/hex"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/google/stenographer/config"
+)
+
+const (
+	baseDir       = "/threadtest/"
+	pktDir        = "/threadtest/pkt/"
+	idxDir        = "/threadtest/idx/"
+	testBlockFile = "../testdata/PKT0/dhcp"
+	testIndexFile = "../testdata/IDX0/dhcp"
+)
+
+func createThreads(t *testing.T, tempDir string) []*Thread {
+	var tc = []config.ThreadConfig{
+		{tempDir + pktDir, tempDir + idxDir, 10, 10},
+	}
+	threads, err := Threads(tc, tempDir+baseDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return threads
+}
+
+func copyData(t *testing.T, tempDir string) {
+	dirs := [...]string{tempDir + pktDir, tempDir + idxDir}
+	testdata := map[string]string{
+		testBlockFile: tempDir + pktDir,
+		testIndexFile: tempDir + idxDir,
+	}
+	for _, dir := range dirs {
+		err := os.MkdirAll(dir, os.FileMode(0755))
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	for src, dst := range testdata {
+		cp := exec.Command("cp", "-rf", src, dst)
+		if err := cp.Run(); err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func rmData(t *testing.T, tempDir string) {
+	os.RemoveAll(tempDir + baseDir)
+}
+
+type requestTest func(
+	method string,
+	path string,
+	headers map[string]string,
+	body string,
+) *httptest.ResponseRecorder
+
+func newRequestTest(t *testing.T, handle http.Handler) requestTest {
+	return func(
+		method string,
+		path string,
+		headers map[string]string,
+		body string,
+	) *httptest.ResponseRecorder {
+
+		req, err := http.NewRequest(
+			method,
+			path,
+			strings.NewReader(body),
+		)
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+		for key, value := range headers {
+			req.Header.Set(key, value)
+		}
+		w := httptest.NewRecorder()
+		handle.ServeHTTP(w, req)
+		return w
+	}
+}
+
+func TestExportDebugHandlers(t *testing.T) {
+	tempDir := os.TempDir()
```

Comment:
ah, sorry, I initially lied to you.

Use http://golang.org/pkg/io/ioutil/#TempDir here instead.

Otherwise LGTM.